### PR TITLE
Fix 625

### PR DESCRIPTION
**Pull request by thomasballinger**
fixing #625 by not adding encoding lines to source files.

### CODE REVIEW

**Comment by thomasballinger** [comment]
Add a new default value for `runsource(encode='auto')`, which uses whether the filename is None to determine whether input is coming from an interactive interface and therefore needs to have an encoding comment added.

Add test for every combination of (unicode/bytestring x encoding comment/none x encode=True/False/'auto'). Some combinations are invalid and raise ValueErrors.

File: bpython/inspection.py [hunk 1]
```diff
@@ -284,6 +284,7 @@ def is_callable(obj):
 
 
 get_encoding_re = LazyReCompile(r'coding[:=]\s*([-\w.]+)')
+get_encoding_line_re = LazyReCompile(r'^.*coding[:=]\s*[-\w.]+.*$')
```
Comment (sebastinas) [inline bpython/inspection.py]: So why do we need that twice?

Comment (thomasballinger) [inline bpython/inspection.py]: I can combine these

**Comment by sebastinas** [comment]
LGTM
